baras/deployments.go

```diff
@@ -349,6 +349,75 @@ var _ = Describe("deployments", func() {
 				return counter
 			}).Should(Equal(10))
 		})
+
+		Context("with instance steps", func() {
+			It("deploys an app, transitions to pause and can be continued multiple times and then deploys successfully", func() {
+				By("Pushing a canary deployment")
+				Eventually(func() string {
+					return helpers.CurlAppRoot(Config, appName)
+				}).Should(ContainSubstring("Hi, I'm Dora"))
+
+				_, originalWorkerStartEvent := GetLastAppUseEventForProcess("worker", "STARTED", "")
+
+				instanceWeights := []int{1, 50, 75, 99}
+
+				deploymentGuid := CreateCanaryDeploymentWithWeightsForDroplet(appGUID, secondDropletGuid, instanceWeights)
+				Expect(deploymentGuid).ToNot(BeEmpty())
+
+				Eventually(func() int { return len(GetProcessGuidsForType(appGUID, "web")) }, Config.CfPushTimeoutDuration()).
+					Should(BeNumerically(">", 1))
+
+				for i := 1; i <= instances; i++ {
```

Comment:
should this be `instanceWeights.Length` (or however you do that in Go).  It's just a coincidence that `instances` == 4, right?